dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -318,45 +365,476 @@ private List<Event> inlineAtomicEnd(FunctionCall ignored) {
         return List.of(EventFactory.Svcomp.newEndAtomic(checkNotNull(currentAtomicBegin)));
     }
 
+    private List<Event> inlinePthreadEqual(FunctionCall call) {
+        final Register resultRegister = getResultRegisterAndCheckArguments(2, call);
+        final Expression leftId = call.getArguments().get(0);
+        final Expression rightId = call.getArguments().get(1);
+        return List.of(
+                EventFactory.newLocal(resultRegister, expressions.makeEQ(leftId, rightId))
```

Comment:
You are missing a cast to the type of `resultRegister` which causes an exception.